File: Sources/CountriesFlagsObjC/source/Countries/UIImage+Honduras.m
//
//  UIImage+Honduras.m
//  CountriesFlags
//

#import "Countries/UIImage+Honduras.h"

#import "Extensions/UIBezierPath+Star.h"
#import "Extensions/CGSize+Extension.h"

@implementation UIImage (Honduras)

+ (UIImage *)flagHondurasWithSize:(CGSize)size {
    UIColor* colorBlueTurquoise = [UIColor colorWithRed:0.0f
                                                  green:188.0f/255.0f
                                                   blue:228.0f/255.0f
                                                  alpha:1.0f];
    UIColor* colorWhite = [UIColor whiteColor];
    
    UIGraphicsBeginImageContextWithOptions(size, YES, 0.0f);
    
    [colorBlueTurquoise setFill];
    UIRectFill(CGRectMake(0.0f, 0.0f, size.width, size.height));
    
    [colorWhite setFill];
    UIRectFill(CGRectMake(0.0f, size.height / 36.0f* 12.0f, size.width, size.height / 36.0f * 12.0f));
    
    [colorBlueTurquoise setFill];
    CGFloat radius = (size.height / 72.0 * 8.0) / 2.0;
    // center
    [[UIBezierPath starAtPoint:CGSizeMidPoint(size)
                        radius:radius
                          type:FlagStarTypeFivePointed] fill];
    // top left
    [[UIBezierPath starAtPoint:CGPointMake(size.width / 72.0f * 26.0f, size.height / 36.0f * 15.0f)
                        radius:radius
                          type:FlagStarTypeFivePointed] fill];
    // top right
    [[UIBezierPath starAtPoint:CGPointMake(size.width / 72.0f * 46.0f, size.height / 36.0f * 15.0f)
                        radius:radius
                          type:FlagStarTypeFivePointed] fill];
    // bottom left
    [[UIBezierPath starAtPoint:CGPointMake(size.width / 72.0f * 26.0f, size.height / 36.0f * 21.0f)
                        radius:radius
                          type:FlagStarTypeFivePointed] fill];
    // bottom right
    [[UIBezierPath starAtPoint:CGPointMake(size.width / 72.0f * 46.0f, size.height / 36.0f * 21.0f)
                        radius:radius
                          type:FlagStarTypeFivePointed] fill];
    
    UIImage* image = UIGraphicsGetImageFromCurrentImageContext();
    UIGraphicsEndImageContext();
    
    return image != nil ? image : [UIImage new];
}

@end

File: Sources/CountriesFlags/Countries/UIImage+Honduras.swift
//
//  UIImage+Honduras.swift
//  CountriesFlags
//

#if canImport(UIKit)

import UIKit

extension UIImage {
    /**
     [wikipedia]: https://en.wikipedia.org/wiki/Flag_of_Honduras "wikipedia"
     
     Generates the flag of Honduras.
     For more information see [wikipedia].
     
     - Parameters:
        - size: image size.
     
     - Returns: Returns the flag of Honduras.
     */
    internal static func flagHonduras(with size: CGSize) -> UIImage {
        let colorBlueTurquoise = UIColor(red: 0.0,
                                         green: 188.0 / 255.0,
                                         blue: 228.0 / 255.0,
                                         alpha: 1.0)
        let colorWhite = UIColor.white

        UIGraphicsBeginImageContextWithOptions(size, true, 0.0)

        colorBlueTurquoise.setFill()
        UIRectFill(CGRect(origin: .zero, size: size))

        colorWhite.setFill()
        UIRectFill(CGRect(origin: CGPoint(x: 0, y: size.height / 36.0 * 12.0),
                          size: CGSize(width: size.width, height: size.height / 36.0 * 12.0)))

        colorBlueTurquoise.setFill()
        let radius = CGFloat(size.height / 72.0 * 8.0) / 2.0
        // center
        UIBezierPath.star(at: size.midPoint, radius: radius).fill()
        // top left
        UIBezierPath.star(at: CGPoint(x: size.width / 72.0 * 26.0, y: size.height / 36.0 * 15.0),
                          radius: radius).fill()
        // top right
        UIBezierPath.star(at: CGPoint(x: size.width / 72.0 * 46.0, y: size.height / 36.0 * 15.0),
                          radius: radius).fill()
        // bottom left
        UIBezierPath.star(at: CGPoint(x: size.width / 72.0 * 26.0, y: size.height / 36.0 * 21.0),
                          radius: radius).fill()
        // bottom right
        UIBezierPath.star(at: CGPoint(x: size.width / 72.0 * 46.0, y: size.height / 36.0 * 21.0),
                          radius: radius).fill()

        let image = UIGraphicsGetImageFromCurrentImageContext()
        UIGraphicsEndImageContext()

        return image ?? UIImage()
    }
}

#endif
